components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -21,10 +22,54 @@ const sessionId = v4();
 const WorkspaceLogs = React.lazy(() => import('../components/WorkspaceLogs'));
 
 export interface StartWorkspaceProps {
-  workspaceId: string;
+  workspaceId: string,
+  parameters?: StartWorkspaceParameters,
+}
+
+export interface StartWorkspaceParameters {
+  trigger?: StartWorkspaceTrigger,
+}
+
+const StartWorkspaceTriggers = {
+  /**
+   * A workspace cluster's ws-proxy redirected the client to "/start/#<wsId>" because it could not find that workspace
+   */
+  "redirect_from_ws_cluster": undefined,
```

Comment:
Semehow it feels like tight coupling of components.

Maybe we shoudl rather have here a param like `autostart=false`?

Reply:
> Semehow it feels like tight coupling of components.

Yes, but `autostart` as well: IMO `ws-proxy`/`supervisor` should not directly control start behavior.

What about `?startError=true`? That would:
 - solve the tight coupling
 - simplify the code :upside_down_face:

Reply:
BTW the specific labels where super helpful for debugging, that was the original intention. But we sure don't need to keep them. :+1:

Reply:
> IMO ws-proxy/supervisor should not directly control start behavior.

Yeah, I was also wondering there should be better way. But could not figure out still how to pass instance id to supervisor frontend yet :(